pysatMagVect/tests/test_comparison.py

```diff
@@ -0,0 +1,212 @@
+                                                                                    -apex_mer[2,:])
+
+        apex_zon[0,:], apex_zon[1,:], apex_zon[2,:] = pysatMagVect.normalize_vector(apex_zon[0,:],
+                                                                                    apex_zon[1,:],
+                                                                                    apex_zon[2,:])
+        apex_fa[0,:], apex_fa[1,:], apex_fa[2,:] = pysatMagVect.normalize_vector(apex_fa[0,:],
+                                                                                apex_fa[1,:],
+                                                                                apex_fa[2,:])
+
+        # calculat mag unit vectors in ECEF coordinates
+        out  = pysatMagVect.calculate_mag_drift_unit_vectors_ecef(eq_lat, eq_long, eq_z,
+                                            [self.inst.date]*len(eq_z))
+        zx, zy, zz, fx, fy, fz, mx, my, mz = out
+
+        # convert into north, east, and up system
+        ze, zn, zu = pysatMagVect.ecef_to_enu_vector(zx, zy, zz, eq_lat, eq_long)
+        fe, fn, fu = pysatMagVect.ecef_to_enu_vector(fx, fy, fz, eq_lat, eq_long)
+        me, mn, mu = pysatMagVect.ecef_to_enu_vector(mx, my, mz, eq_lat, eq_long)
+
+
+        # create inputs straight from IGRF
+        igrf_n = []
+        igrf_e = []
+        igrf_u = []
+        for lat, lon, alt in zip(eq_lat, eq_long, eq_z):
+            out = pysatMagVect.igrf.igrf13syn(0, date.year, 1, alt,
+                                            np.deg2rad(90-lat), np.deg2rad(lon))
+            out = np.array(out)
+            # normalize
+            out /= out[-1]
+            igrf_n.append(out[0])
+            igrf_e.append(out[1])
+            igrf_u.append(-out[2])
+
+
+        # # geodetic to geographic
+        # x, y, z = pysatMagVect.geodetic_to_ecef(data['glat'], data['glong'], data['alt'])
+        # data['glat'], data['glong'], data['alt'] = pysatMagVect.ecef_to_geocentric(x, y, z)
+        #
+        # # test enu and ecef conversions
+        # igrf_e = np.array(igrf_e)
+        # igrf_n = np.array(igrf_n)
+        # igrf_u = np.array(igrf_u)
+        # x, y, z = pysatMagVect.enu_to_ecef_vector(igrf_e, igrf_n, igrf_u, data['glat'],
+        #                                           data['glong'])
+        # igrf_e2, igrf_n2, igrf_u2 = pysatMagVect.ecef_to_enu_vector(x, y, z, data['glat'],
+        #                                                             data['glong'])
+        # # they check out! :)
+
+
+        # dot product of zonal and field aligned
+        dot_apex_zonal = apex_fa[0,:]*apex_zon[0,:] + apex_fa[1,:]*apex_zon[1,:] + apex_fa[2,:]*apex_zon[2,:]
+        dot_apex_mer = apex_fa[0,:]*apex_mer[0,:] + apex_fa[1,:]*apex_mer[1,:] + apex_fa[2,:]*apex_mer[2,:]
+
+        dotmagvect_zonal = ze*fe + zn*fn + zu*fu
+        dotmagvect_mer = me*fe + mn*fn + mu*fu
+
+
+        try:
```

Comment:
Is making a figure the most efficient way to do a unit test?

Reply:
For this package unit test plots are the way to go. I've used every plot to get the code to where it is at.

Reply:
Added issue #38 to ensure this is resolved.